xrootd/xrootd_config.go

```diff
@@ -605,6 +607,67 @@ func LaunchXrootdMaintenance(ctx context.Context, server server_structs.XRootDSe
 	)
 }
 
+// The default config has `Xrootd.AuthRefreshInterval: 5m`, which we need to convert
+// to an integer representation of seconds for our XRootD configuration. This hook
+// handles that conversion during unmarshalling, as well as some sanitization of user inputs.
+func authRefreshStrToSecondsHookFunc() mapstructure.DecodeHookFuncType {
```

Comment:
I've done a similar string to xrootd int value conversion for cache watermark parameter: https://github.com/PelicanPlatform/pelican/blob/c5ad188c8f1c43d2265843fb37dde3459933e1b0/config/config.go#L802-L839

I didn't do it at the marshaling level but in the config validation, where I read in the value, check the value, and return the xrootd value if everything goes well. At some point we should consolidate different approaches into a unified one.

One repetition I saw here is handling the duration conversion. Viper has the nice check against the config value and we should be able to reuse that and get the value in second instead. I'll let you decide

Reply:
I think I'd rather leave this in the unmarshal hook for now, since it handles type conversion of `time.Duration --> int`.